src/actions/index.ts

```diff
@@ -7,7 +7,8 @@ export const addFigure = (type : string, x: number, y: number) => {
             id: nextFigureId++,
             type: type,
             x: x,
-            y: y
+            y: y,
+            selected: false
```

Comment:
i'd like to see a more defined model for your application. there's a canvas, canvas has elements of type shape rectangle, triangle, circle, etc.